include/turtle/detail/parameter.hpp

```diff
@@ -10,24 +10,70 @@
 #define MOCK_PARAMETER_HPP_INCLUDED
 
 #include "../config.hpp"
-#include <boost/function_types/parameter_types.hpp>
-#include <boost/function_types/function_arity.hpp>
-#include <boost/mpl/at.hpp>
 
 namespace mock
 {
 namespace detail
 {
+    template< class... >
+    struct tuple;
+
+    template< std::size_t I, class T >
+    struct tuple_element;
+ 
+    template< std::size_t I, class H, class... T >
+    struct tuple_element<I, tuple<H, T...>> : tuple_element<I-1, tuple<T...>>
+    {};
+ 
+    template< class H, class... T >
+    struct tuple_element<0, tuple<H, T...>>
+    {
+       using type = H;
+    };
+
+    template< typename Signature >
+    struct result_type;
+
+    template< typename R, typename... Args >
+    struct result_type< R(Args...) >
+    {
+        using type = R;
+    };
+
+    template< typename Signature >
+    struct function_arity;
+
+    template< typename R, typename... Args >
+    struct function_arity< R(Args...) >
+    {
+        static constexpr size_t value = sizeof...(Args);
+    };
+
+    template< typename Signature >
+    struct parameter_types;
+
+    template< typename R, typename... Args >
+    struct parameter_types< R(Args...) >
+    {
+        using type = tuple<Args...>;
+    };
+
     template< typename Signature, int n >
     struct parameter
     {
-        typedef typename
-            boost::mpl::at_c<
-                typename
-                    boost::function_types::parameter_types< Signature >,
-                n
-            >::type type;
+        static_assert(n < function_arity<Signature>::value, "Function signature has not that many parameters");
+        using type = typename tuple_element< n, typename parameter_types<Signature>::type >::type;
+    };
+
+    template<typename T>
+    struct parameter_type;
+    template<typename T, typename U>
+    struct parameter_type<T(U)>
+    {
+        using type = U;
     };
+    template<typename T>
+    using parameter_type_t = typename parameter_type<T>::type;
```

Comment:
Would it make sense to use boost::callable_traits for parameter type deduction instead?

Reply:
Would be possible. But similar to `is_callable` this only supports simple methods which seemingly is(/was?) enough, so this (much smaller) implementation should suffice.
And Boost.CallableTraits was introduced in Boost 1.66, i.e. it would need another increase in the minimum Boost version required which I wanted to avoid at the time of writing this.